src/listener/Image.php

```diff
@@ -20,11 +20,12 @@ class Image extends InlineListener
     /**
      * {@inheritDoc}
      */
-    public function process(Line $line)
+    public function process(Line $line, Lexer $lexer=null)
     {
         $embedUrl = $line->insertJsonKey('image');
         if ($embedUrl) {
-            $this->updateInput($line, str_replace(['{src}'], [self::escape($embedUrl)], $this->wrapper));
+            $embedUrl = ($lexer->escapeInput) ? self::escape($embedUrl) : $embedUrl;
```

Comment:
Same here, as we strictly know what value $mebedUrl must contain, we can just encode the input without checking whether enabled or not.

Reply:
(same for the other checks in the other listeners - we know the types, encoding is safe as we know the content)